scene/gui/base_button.cpp

```diff
@@ -282,10 +282,7 @@ void BaseButton::_notification(int p_what) {
 	if (p_what == NOTIFICATION_ENTER_TREE) {
 	}
 
-	if (p_what == NOTIFICATION_EXIT_TREE) {
-	}
-
-	if (p_what == NOTIFICATION_VISIBILITY_CHANGED && !is_visible_in_tree()) {
+	if (p_what == NOTIFICATION_EXIT_TREE || p_what == NOTIFICATION_VISIBILITY_CHANGED && !is_visible_in_tree()) {
```

Comment:
scene/gui/base_button.cpp: In member function 'void BaseButton::_notification(int)':
scene/gui/base_button.cpp:285:84: error: suggest parentheses around '&&' within '||' [-Werror=parentheses]
 NOTIFICATION_EXIT_TREE || p_what == NOTIFICATION_VISIBILITY_CHANGED && !is_visible_in_tree()) {
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~